sunpy/instr/goes.py

```diff
@@ -260,8 +277,8 @@ def goes_chianti_tem(longflux, shortflux, satellite=8,
 
     """
     # ENSURE INPUTS ARE OF CORRECT TYPE AND VALID VALUES
-    longflux = np.asanyarray(longflux, dtype=np.float64)
-    shortflux = np.asanyarray(shortflux, dtype=np.float64)
+    longflux.to(u.W/u.m/u.m)
```

Comment:
I think this returns a new array, rather than modifying in place. What are you trying to achieve here?